libs/@blockprotocol/type-system/rust/src/ontology/id/mod.rs

```diff
@@ -217,37 +219,193 @@ impl<'a> FromSql<'a> for BaseUrl {
+        /// when multiple parties are drafting versions targeting the same published version.
+        lane: String,
+        /// Monotonic revision number within this lane.
+        ///
+        /// Should increment with each new draft in the same lane. Revisions are
+        /// compared numerically when ordering drafts in the same lane.
+        revision: u32,
+    },
+}
+
+impl FromStr for PreRelease {
+    type Err = ParseDraftInfoError;
+
+    fn from_str(draft_info: &str) -> Result<Self, Self::Err> {
+        draft_info.rsplit_once('.').map_or(
+            Err(ParseDraftInfoError::IncorrectFormatting),
+            |(lane, revision)| {
+                Ok(Self::Draft {
+                    lane: lane.to_owned(),
+                    revision: u32::from_str(revision).map_err(|error| {
+                        if *error.kind() == IntErrorKind::Empty {
+                            ParseDraftInfoError::MissingRevision
+                        } else {
+                            ParseDraftInfoError::InvalidRevision(
+                                revision.to_owned(),
+                                error.to_string(),
+                            )
+                        }
+                    })?,
+                })
+            },
+        )
+    }
+}
+
+/// A semantic version number for ontology types, supporting both published and pre-release
+/// versions.
+///
+/// [`OntologyTypeVersion`] represents the version component of a [`VersionedUrl`], following
+/// semantic versioning conventions. It supports:
+///
+/// - **Published versions**: Simple major version numbers (e.g., `1`, `2`, `3`)
+/// - **Pre-release versions**: Major version with pre-release metadata (e.g., `2-draft.lane.5`)
+///
+/// # Ordering Semantics
+///
+/// Versions follow semantic versioning rules:
+/// - Major version takes precedence: `v/3` > `v/2` > `v/1`
+/// - Published versions have higher precedence than pre-release versions with the same major
+///   version
+/// - Pre-release versions are ordered lexicographically by lane, then numerically by revision
+///
+/// # Examples
+///
+/// ```
+/// use std::str::FromStr;
+///
+/// use type_system::ontology::id::{OntologyTypeVersion, PreRelease};
+///
+/// // Published version
+/// let v1 = OntologyTypeVersion::from_str("1").unwrap();
+/// assert_eq!(v1.major, 1);
+/// assert!(v1.pre_release.is_none());
+///
+/// // Draft version
+/// let draft = OntologyTypeVersion::from_str("2-draft.lane123.5").unwrap();
+/// assert_eq!(draft.major, 2);
+/// assert!(matches!(draft.pre_release, Some(PreRelease::Draft { .. })));
+///
+/// // Ordering: published > draft (same major)
+/// assert!(v1 < draft); // v/1 < v/2-draft.lane123.5
+/// let v2 = OntologyTypeVersion::from_str("2").unwrap();
+/// assert!(draft < v2); // v/2-draft.lane123.5 < v/2
+/// ```
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+#[cfg_attr(feature = "codegen", derive(specta::Type), specta(transparent))]
+pub struct OntologyTypeVersion {
```

Comment:
Again, would it make sense to simply wrap `semver::Version` here and ensure that the `minor` and `patch` are always `0`?

Reply:
This would mean that we wouldn't need to implement ordering, as ordering is defined per semver spec

Reply:
But it does, semver orders lexographically, which means that the ordering would be consistent. It would be _exactly_ the same as what you're doing right now.

11.4:

> Precedence for two pre-release versions with the same major, minor, and patch version MUST be determined by comparing each dot separated identifier from left to right until a difference is found as follows:
> 1. Identifiers consisting of only digits are compared numerically.
> 2. Identifiers with letters or hyphens are compared lexically in ASCII sort order.
> 3. Numeric identifiers always have lower precedence than non-numeric identifiers.
> 4. A larger set of pre-release fields has a higher precedence than a smaller set, if all of the preceding identifiers are equal.